charts/partials/templates/_network-validator.tpl

```diff
@@ -5,16 +5,7 @@ imagePullPolicy: {{.Values.proxy.image.pullPolicy | default .Values.imagePullPol
 {{ include "partials.resources" .Values.proxy.resources }}
 {{- if or .Values.networkValidator.enableSecurityContext }}
 securityContext:
-  allowPrivilegeEscalation: false
-  capabilities:
-    drop:
-    - ALL
-  readOnlyRootFilesystem: true
-  runAsGroup: 65534
-  runAsNonRoot: true
-  runAsUser: 65534
-  seccompProfile:
-    type: RuntimeDefault
+  {{- toYaml .Values.networkValidator.securityContext | trim | nindent 2 }}
```

Comment:
I think we have to be careful here with how Helm does value preserving upgrades.  Since the securityContext value didn't exist in the previous version, when upgrading I believe that the zero value will get used for it instead of the new default in the new version.

We probably need something like `default .Values.networkValidator.securityContext <default value here>`

Reply:
yes, this looks right to me.  If securityContext is set, we'll use what's been set.  If it's not set, we'll use the default from the else branch.  Might be worth testing the Helm upgrade flow to verify that our understanding of this is correct.